### DISCUSSION TITLE
Can you use the TPM on your Linux Surface?

### DISCUSSION THREAD

**Discussion by Gefrierbox** [linux-surface/linux-surface#880] [Q&A]
I'm using a Surface Pro 6 and like to install Linux on it. But I also want to encrypt my storage using the TPM. 
Has anyone tried using the TPM chip on your Linux Surface and could share his/her experience? I would try it myself but I'm afraid that the TPM will stop working as soon as I install a Linux distribution (like a comment said here: https://docs.microsoft.com/en-us/answers/questions/599174/unable-to-enable-tpm-on-surface-book-2-tpm-failure.html or here (german): https://www.drwindows.de/news/gastbeitrag-linux-auf-dem-surface-ein-erfahrungsbericht).
Any information regarding the TPM would be apperciated.

**Comment by StollD** [comment]
I did use the TPM for encryption a few years ago. It worked fine, but the software side was not quite there, so there were some quirks. Since systemd has gained TPM functionality iirc, it should be much better these days.

> I would try it myself but I'm afraid that the TPM will stop working as soon as I install a Linux distribution

TPMs work by binding a value to a state of the TPM. That state is modified by the programs that run during the boot process. If the state changes, the value that is bound to a different state can not be unlocked. Things like disabling or enabling secure boot will change the state of the TPM.

When you install Linux the TPM wont stop working, but its state might change. This means that Bitlocker wont be able to read your disk anymore and Windows wont boot without the recovery key. But this depends on how you boot Windows. If you boot it from the UEFI entry, it will be fine, since the programs that run and their order are not changed (UEFI -> Windows Bootloader -> Windows). But if you chainload Windows from GRUB, the programs and their order will be different (UEFI -> shim -> grub -> Windows Bootloader -> Windows).

For using it with Linux the only thing that matters is getting the state of the TPM right. Install the distribution, enable secureboot, check if everything works, then add a LUKS key that you bind to the TPM with something like systemd-cryptsetup. Make sure that you have a second LUKS key that you can use to unlock the disk if the state of the TPM changes.
